### DISCUSSION TITLE
Non TRF vs TRF pre-trained model discussion

### DISCUSSION THREAD

**Discussion by nlovell1** [explosion/spaCy#10929] [Help: Coding & Implementations]
Hi, a few things
1. What are the differences between a non-TRF [es_core_news_lg](https://spacy.io/models/es#es_core_news_lg), for example and TRF [es_dep_news_trf](https://spacy.io/models/es#es_dep_news_trf) model behind the scenes? Looking for an in depth answer, couldn't find it on the docs (maybe looking in the wrong place)?
2. Additionally, why doesn't the TRF model include `ner`, and what's the most optimal way to get named entity recognition while still retaining the tokenization/lemmatization accuracy of the TRF model?

**Comment by polm** [comment]
> What are the differences between a non-TRF [es_core_news_lg](https://spacy.io/models/es#es_core_news_lg), for example and TRF [es_dep_news_trf](https://spacy.io/models/es#es_dep_news_trf) model behind the scenes? Looking for an in depth answer, couldn't find it on the docs (maybe looking in the wrong place)?

The `trf` model uses Transformers, the `lg` model doesn't. The models are defined by their training data and components as described on the page you linked to. 

> Additionally, why doesn't the TRF model include ner, and what's the most optimal way to get named entity recognition while still retaining the tokenization/lemmatization accuracy of the TRF model?

I forget exactly why the `trf` pipeline doesn't include NER - I think it was to keep model size down or something (someone else on the team can correct me on that). But the tokenizer and lemmatizer in the Spanish pipelines are rule-based, so whether the model is TRF or not doesn't matter.

If you have data then you can just train a custom NER model and source the lemmatizer/tagger from another pipeline, see [the sourcing docs](https://spacy.io/usage/processing-pipelines#sourced-components).
